core/src/main/java/org/infinispan/remoting/rpc/RpcOptions.java

```diff
@@ -111,21 +60,14 @@ public boolean equals(Object o) {
 
       if (timeout != options.timeout) return false;
       if (deliverOrder != options.deliverOrder) return false;
-      if (responseFilter != null ? !responseFilter.equals(options.responseFilter) : options.responseFilter != null)
-         return false;
-      if (responseMode != options.responseMode) return false;
-      if (unit != options.unit) return false;
-
-      return true;
+      return unit == options.unit;
```

Comment:
nitpick: replace `if` with `&&`
```suggestion
      return timeout == options.timeout && 
             deliverOrder == options.deliverOrder && 
             unit == options.unit;
```